delfin/drivers/hpe/hpe_3par/component_handler.py

```diff
@@ -417,3 +417,245 @@ def parse_speed(self, speed_value):
+                    "description": volume_group.get('comment'),
+                    "storage_id": storage_id,
+                    "native_volume_group_id": volume_group.get('id'),
+                    "volumes": ','.join(volume_ids)
+                }
+                volume_group_list.append(volume_group_model)
+            volume_group_relation_list = []
+            for volume_group in volume_group_list:
+                volumes = volume_group.pop('volumes', None)
+                if not volumes:
+                    continue
+                volumes = volumes.split(',')
+
+                for volume in volumes:
+                    volume_group_relation = {
+                        'storage_id': storage_id,
+                        'native_volume_group_id':
+                            volume_group.get('native_volume_group_id'),
+                        'native_volume_id': volume}
+                    volume_group_relation_list.append(volume_group_relation)
+
+            result = {
+                'volume_groups': volume_group_list,
+                'vol_grp_vol_rels': volume_group_relation_list
+            }
+        return result
+
+    def list_masking_views(self, storage_id):
+        views = self.ssh_handler.list_masking_views()
+        views_list = []
+        if views:
+            hosts_map = self.ssh_handler.get_resources_ids(
+                self.ssh_handler.HPE3PAR_COMMAND_SHOWHOST_D,
+                consts.HOST_OR_VV_PATTERN)
+            hosts_group_map = self.ssh_handler.get_resources_ids(
+                self.ssh_handler.HPE3PAR_COMMAND_SHOWHOSTSET_D,
+                consts.HOST_OR_VV_PATTERN)
+            volumes_map = self.ssh_handler.get_resources_ids(
+                self.ssh_handler.HPE3PAR_COMMAND_SHOWVV,
+                consts.HOST_OR_VV_PATTERN)
+            volumes_group_map = self.ssh_handler.get_resources_ids(
+                self.ssh_handler.HPE3PAR_COMMAND_SHOWVVSET_D,
+                consts.HOST_OR_VV_PATTERN)
+            for view in views:
+                vv_name = view.get('vvname')
+                host_name = view.get('hostname')
+                if vv_name and host_name:
+                    port = view.get('port', '').replace('-', '')
+                    id = view.get('lun')
+                    wwn = view.get('host_wwn/iscsi_name', '').replace('-', '')
+                    native_port_group_id = None
+                    if port:
+                        id = '%s_%s' % (view.get('lun'), port)
+                        native_port_group_id = 'port_group_%s' % port
+                    if wwn:
+                        id = '%s_%s' % (id, wwn)
+                    view_model = {
+                        'native_masking_view_id': id,
+                        "name": view.get('lun'),
+                        'native_port_group_id': native_port_group_id,
+                        "storage_id": storage_id
+                    }
+                    if 'set:' in vv_name:
```

Comment:
We need to validate mandatory fields are set in masking view.

From file `<>/delfin/drivers/driver.py`, function `list_masking_views()` comments, mandatory fields:

From host side: Mandatorily one of the (native_storage_host_group_id  | native_storage_host_id)
From volume side: Mandatorily one of the (native_volume_group_id | native_volume_id)
From port side: Optionally (native_port_group_id)

Reply:
Modified